airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/interpolated_request_input_provider.py

```diff
@@ -2,27 +2,35 @@
 # Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 #
 
-from typing import Any, Mapping, MutableMapping
+from typing import Any, Mapping, Optional, Union
 
 from airbyte_cdk.sources.declarative.interpolation.interpolated_mapping import InterpolatedMapping
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.interpolation.jinja import JinjaInterpolation
 
 
 class InterpolatedRequestInputProvider:
     """
-    Helper class that generically performs string interpolation on the provided dictionary input
+    Helper class that generically performs string interpolation on the provided dictionary or string input
     """
 
     def __init__(self, *, config, request_inputs=None):
+        self._config = config
+
         if request_inputs is None:
             request_inputs = {}
-        self._interpolator = InterpolatedMapping(request_inputs, JinjaInterpolation())
-        self._config = config
+        if isinstance(request_inputs, str):
+            self._interpolator = InterpolatedString(request_inputs, "")
+        else:
+            self._interpolator = InterpolatedMapping(request_inputs, JinjaInterpolation())
 
     def request_inputs(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
+    ) -> Optional[Union[Mapping, str]]:
```

Comment:
Ah nice catch yeah it's probably unnecessary, we can just return the union. I think I originally put it in since the new json and data body methods returned optionals, but those can just be wrapped there.